test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -165,78 +244,75 @@ def read_register(self, dut, destination, source, reg_width, address,
             dut:            device under test.
             destination:    id of the target module.
             source:         id of the sending module.
-            reg_width:      choose between 16, 32, 64 and 128 bit register
+            word_width:     choose between 16, 32, 64 and 128 bit register
                             access.
             address:        address of the register the value is to be read
                             from.
-            response:       buffer variable which the response of the dut will
-                            be stored in
+            rx_packet:      debug interconnect packet which will mirror what the
+                            dut replied with
 
         Returns:
-            -
+            response of the debug module of variable length in the rx_packet
 
         """
 
         # branch depending on the width of the accessed data word
-        if reg_width == 16:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_16
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16
-
-        elif reg_width == 32:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_32
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32
-
-        elif reg_width == 64:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_64
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64
-
-        elif reg_width == 128:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_128
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128
-
+        if word_width == 16:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_16
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16
+        elif word_width == 32:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_32
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32
+        elif word_width == 64:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_64
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64
+        elif word_width == 128:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_128
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128
         else:
-            # display an error if an unsupported register width was chosen
-            raise TestFailure("Invalid register width parameter")
 
-        packet = DiPacket(destination = destination, source = source,
-                          packet_type = self.PACKET_TYPE_REG,
-                          packet_subtype = subtype, payload = address)
+            # display an error if an unsupported register width was chosen
```

Comment:
Add comments to code where the code doesn't speak for itself, but don't add them otherwise. In this case, the error message says exactly the same thing as the comment, i.e. the comment is unnecessary. Actually, it's even harmful: it makes your code longer, i.e. it takes longer to read and it in subsequent changes, comments can easily get out of date.